semantics/src/traversal/functions.rs

```diff
@@ -79,6 +83,38 @@ pub fn func_def(
     unreachable!()
 }
 
+fn validate_all_paths_return_or_revert(
+    body: &[Spanned<fe::FuncStmt>],
+) -> Result<(), SemanticError> {
+    // This will need to become more sophisticated when we introduce branching logic
+    // because we then need to follow different code paths and check that they
+    // all return or revert.
+    for statement in body {
+        if let fe::FuncStmt::Return { .. } = &statement.node {
+            return Ok(());
+        }
+
+        if let fe::FuncStmt::Revert { .. } = &statement.node {
+            return Ok(());
+        }
+    }
+
+    Err(SemanticError::MissingReturn)
+}
+
+fn validate_no_path_returns(body: &[Spanned<fe::FuncStmt>]) -> Result<(), SemanticError> {
```

Comment:
This rule will probably become obsolete when we introduce the implicit return of `()` because I believe such a method will then be perfectly legal:

```
def no_return_in_signature():
    return ()
```

The same thing is legal in rust:

```rust
fn no_return() {
    return ();
}
```